formstack-baton-requests/src/main/scala/com/gu/identity/formstackbatonrequests/services/FormstackService.scala

```diff
@@ -141,25 +171,73 @@ object FormstackService extends FormstackRequestService with LazyLogging {
+    } yield ValidatedSubmissions(
+      accountOneResponse = accountOneResults,
+      accountTwoResponse = accountTwoResults)
+  }
+
   override def submissionData(submissionIdEmails: List[SubmissionIdEmail], config: PerformLambdaConfig): Either[Throwable, List[FormstackSubmissionQuestionAnswer]] = {
     logger.info(s"retrieving submission data for ${submissionIdEmails.length} submissions")
-    val tokens = List(config.accountOneToken, config.accountTwoToken)
-    tokens.traverse { token =>
-      val accountSubmissions = submissionIdEmails.filter(sub => sub.accountNumber == token.account)
-      for {
-        fieldsAndValues <- getSubmissions(accountSubmissions, token, config.encryptionPassword)
-        labelsAndValues <- getSubmissionQuestionsAnswers(fieldsAndValues, token)
-      } yield labelsAndValues
-    }.map(accountSubmissions => accountSubmissions.flatten)
+    val accountTwoSubmissions = submissionIdEmails.filter(_.accountNumber == config.accountTwoToken.account)
+    for {
+      accountTwoResults <- getSubQandAForAccount(accountTwoSubmissions, config.accountTwoToken, config.encryptionPassword)
+      accountOneSubmissions = submissionIdEmails.filter(_.accountNumber == config.accountOneToken.account)
+      submissionsToFetchFromAccountOne = accountOneSubmissions ++ accountTwoResults.notFound
+      accountOneResults <- getSubQandAForAccount(submissionsToFetchFromAccountOne, config.accountOneToken, config.encryptionPassword)
+    } yield accountOneResults.found ++ accountTwoResults.found
   }
 
+  def validateAndFixSubmissionIdEmails(submissionIdEmails: List[SubmissionIdEmail], config: PerformLambdaConfig): Either[Throwable, List[SubmissionIdEmail]] = {
+    val submissionIdEmailbyId: Map[String, SubmissionIdEmail] = submissionIdEmails.map(sidEmail => sidEmail.submissionId -> sidEmail).toMap
+
+    def fixedSubmissionsFor(subResponse: SubmissionsResponse, originAccount: Int): List[SubmissionIdEmail] = {
+      //for each submissions found in this account get the submission and make sure it refers to the correct id
+      subResponse.found.map{ validatedSubmission =>
+        submissionIdEmailbyId(validatedSubmission.id).copy(accountNumber = originAccount)
+      }.toList
+
+    }
+    for {
+      validatedSubmissions <- getValidatedSubmissionData(submissionIdEmails, config)
+    } yield fixedSubmissionsFor(validatedSubmissions.accountOneResponse, 1) ++ fixedSubmissionsFor(validatedSubmissions.accountTwoResponse, 2)
+  }
   override def deleteUserData(submissionIdEmails: List[SubmissionIdEmail], config: PerformLambdaConfig): Either[Throwable, List[SubmissionDeletionReponse]] = {
+    for {
+      //we regenerate the submissionIdEmails to make sure they all refer to the formstack account where the submission was verified to exist
+      fixedSubmissionIdEmails <- validateAndFixSubmissionIdEmails(submissionIdEmails, config)
+      deleteResponse <- deleteValidatedSubmissions(fixedSubmissionIdEmails, config)
```

Comment:
`deleteValidatedSubmissions` is just a rename of the current deleteUserData function.
The idea is to leave the deletion code as unchanged as possible since we cannot test it. This just adds a step at the start to "fix" the list of submissions that we are going to delete. We remove the submissions that are not found anywhere and we change the account id of the submissions that are now in account 1 even though dynamodb says they should be in account 2